### PR TITLE
feat: create modal component

### PR DESCRIPTION
**Pull request by luqven**

### CODE REVIEW

**Comment by luqven** [comment]
Current dependencies on/for this PR:
* next
  * **PR #184** 
    * **PR #185**  (this PR)
      * **PR #187** 
        * **PR #186** 

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: frontend/src/components/layouts/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      window.removeEventListener("keyup", keyHandler);
+    };
+  }, [open, locked, onClose]);
+
+  console.log("[imgix] Modal", open ? "opened" : "closed");
+
+  return (
+    <>
+      {open && (
+        <Portal className="modalPortal">
```
Comment (frederickfogerty) [inline frontend/src/components/layouts/Modal.tsx:49]: Where is this class name defined?

File: frontend/src/components/layouts/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          <div
+            ref={backdrop}
+            className={`${styles.backdrop} ${open && "active"}`}
```
Comment (frederickfogerty) [inline frontend/src/components/layouts/Modal.tsx:52]:
```suggestion
            className={`${styles.backdrop} ${open && styles.active}`}
```

File: frontend/src/components/layouts/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          >
+            <div className={`${styles.content} modal-content`}>{children}</div>
```
Comment (frederickfogerty) [inline frontend/src/components/layouts/Modal.tsx:54]:
```suggestion
            <div className={`${styles.content} ${styles['modal-content']}`}>{children}</div>
```
could also be styles.modalContent, not sure

**Review by frederickfogerty — COMMENTED** [review]
LGTM so far

File: frontend/src/components/layouts/Modal.tsx (hunk 2 above)
Comment (luqven) [reply frontend/src/components/layouts/Modal.tsx:52]: 🤦🏼‍♂️ 100%, nice catch.

File: frontend/src/components/layouts/Modal.tsx (hunk 1 above)
Comment (luqven) [reply frontend/src/components/layouts/Modal.tsx:49]: It'd not being used for anything. This really should be an `id` not a `className`.

File: frontend/src/components/layouts/Modal.tsx (hunk 3 above)
Comment (luqven) [reply frontend/src/components/layouts/Modal.tsx:54]: Yup nice catch forgot the update the class here as well.

**Review by frederickfogerty — CHANGES_REQUESTED** [review]
Seems like the outstanding comments haven't been resolved yet

**Review by frederickfogerty — COMMENTED** [review]
LGTM

**Pull request merged**